WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,147 @@
+
+        <li class="same-line"><a href="#comp-query-form" class="js-scroll-to">Top</a></li>
+      </ul>
+    <% end %>
+  </div>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline", id: "comp-query-form" do %>
+    <%= render 'index_form' %>
+  <% end %>
+
+  <% if @competitions.length == 0 %>
+    <div class="alert alert-warning">No competitions found.</div>
+  <% else %>
+    <% if params[:display] == "List" %>
+      <div class="row competition-list">          
+        <% if @in_progress_competitions.length > 0 %>
+          <div class="col-md-12" id="current">  
+            <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+          </div>
+        <% end %>
+
+        <% if @upcoming_competitions.length > 0 %>
+          <div class="col-md-12" id="upcoming">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+        <% end %>
+
+        <% if @past_competitions.length > 0 %>
+          <div class="col-md-12" id="past">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions (last #{CompetitionsController::PAST_COMPETITIONS_DAYS} days)" %>
+          </div>
+        <% end %>
+      </div>
+    <% else %>
+      <div class="row competition-list">
+        <div class="col-xs-12 col-md-12">
+          <div id="comps-map"></div>
+        </div>
+      </div>
     <% end %>
-  </table>
+  <% end %>
 </div>
+
+<script>
+// See https://github.com/cubing/worldcubeassociation.org/issues/356
+function mapSize() {
+  var nav = parseInt($(".navbar").outerHeight(true));
+  var head = parseInt($("h2").outerHeight(true));
+  var form = parseInt($(".form-group").outerHeight(true));
+  var foot = parseInt($(".footer").outerHeight(true));
+
+  var height = parseInt($(window).innerHeight());
+
+  var map = height - nav - head - form - foot;
+
+  $("#comps-map").height( map );
+}
+
+function affixWidth() {
+  // From https://github.com/twbs/bootstrap/issues/6350#issuecomment-16069663
+  // also http://stackoverflow.com/a/20301984
+  // and http://stackoverflow.com/a/24908611
+  var affixBase = $(".container");
+  var width = affixBase.width() - parseInt(affixBase.css('padding-left')) - parseInt(affixBase.css('padding-right'));
+  $("#simple-nav").width(width);
+}
+
+// From http://stackoverflow.com/a/30966965
+$(".js-scroll-to").click(function(e) {
+  e.preventDefault();
+
+  // Only apply the movement if there is a div to go to.
+  if(!$(this).hasClass('disabled')) {
+    // Little hack to activate the affix thing and
+    // move correctly to the anchor.
+    $("#simple-nav").removeClass('affix-top').addClass('affix');
```

Comment:
This is tricky, and I'm not sure it works all the time. Consider the following scenario.

Fresh page load:

![image](https://cloud.githubusercontent.com/assets/277474/14007630/bf2edaf6-f135-11e5-88db-b0a3be9d99d0.png)


Now click on Top:

![image](https://cloud.githubusercontent.com/assets/277474/14007637/c7a0d27a-f135-11e5-9789-9099670606ed.png)

Now scroll up:

![image](https://cloud.githubusercontent.com/assets/277474/14007641/d0ec51ba-f135-11e5-987c-a2b244dd3410.png)

See how the navbar is staying affixed? That's because the plugin never deactivated the affix, presumably because it should have been deactivated the whole time.

What happens if you just remove this line entirely? Does the affix library not kick in until you explicitly scroll? I think we should just find some way of poking the affix library, rather than trying to do its job for us. See the `checkPosition` method here: http://getbootstrap.com/javascript/#affixcheckposition.